src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestHostLauncher.cs

```diff
@@ -19,5 +21,13 @@ public interface ITestHostLauncher
         /// <param name="defaultTestHostStartInfo">Default TestHost Process Info</param>
         /// <returns>Process id of the launched test host</returns>
         int LaunchTestHost(TestProcessStartInfo defaultTestHostStartInfo);
+
+        /// <summary>
+        /// Launches custom test host using the default test process start info
+        /// </summary>
+        /// <param name="defaultTestHostStartInfo">Default TestHost Process Info</param>
+        /// <param name="cancellationToken">The cancellation Token.</param>
+        /// <returns>Process id of the launched test host</returns>
+        int LaunchTestHost(TestProcessStartInfo defaultTestHostStartInfo, CancellationToken cancellationToken);
```

Comment:
Will require a corresponding change in TPV1 OM. TE implements this for debugging which would need a change as well.

Reply:
This interface is used by run time provider only for launching the custom host via custom host launcher.
Why changes are required in TPV1 OM?

Even though if TE uses it as an interface for their internal purpose, there will no breaking change here (as TPv1 OM is used both while build and run time. TPV2 OM is not loaded.) Please correct me if i am missing something here.

Reply:
@cltshivash @singhsarab @AbhitejJohn 
Am I missing something here? AFIAK ITestHostLauncher is meant only for runtime provider and it is loaded in vstest console only. Neither translation layer, nor test host or any other process from TP side uses this except vstest.console.

If TE is using this interface for their internal purpose, then it is using TPV1 OM and loading TPV1 OM. So how the behavior is inconsistent? I don't feel comfortable changing TPV1 OM unless it's a breaking change.